packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -763,7 +764,19 @@ function applyKey(
             }
         }
 
-        element.directives.push(ast.keyDirective(keyAttribute.value, keyAttribute.location));
+        const parentIsIteration = !!(forOfParent || forEachParent);
+
+        if (!parentIsIteration) {
+            ctx.warnAtLocation(
```

Comment:
You can use `warnOnNode` instead of `warnAtLocation` this would avoid having to pass the `parse5ElmLocation`.

```suggestion
            ctx.warnOnNode(
```

Reply:
Thanks for the tip! Fixed.